server/src/main/java/org/elasticsearch/search/aggregations/bucket/composite/CompositeAggregationFactory.java

```diff
@@ -41,6 +45,13 @@ class CompositeAggregationFactory extends AggregatorFactory {
     @Override
     protected Aggregator createInternal(Aggregator parent, CardinalityUpperBound cardinality, Map<String, Object> metadata)
         throws IOException {
-        return new CompositeAggregator(name, factories, context, parent, metadata, size, sources, afterKey);
+        try {
+            return new CompositeAggregator(name, factories, context, parent, metadata, size, sources, afterKey);
+        } catch (MultiBucketConsumerService.TooManyBucketsException e) {
+            logger.error(
```

Comment:
I don't think this should be an error log. For example when circuit breaker is tripped this is logged as warning.